File: app/src/main/java/com/pr0gramm/app/ui/fragments/PreviewInfoSource.java
package com.pr0gramm.app.ui.fragments;

import android.support.annotation.Nullable;

import com.pr0gramm.app.feed.FeedItem;
import com.pr0gramm.app.ui.PreviewInfo;

/**
 */
public interface PreviewInfoSource {
    /**
     * Returns the preview info for the given feed item.
     */
    @Nullable
    PreviewInfo previewInfo(FeedItem item);
}

File: app/src/main/java/com/pr0gramm/app/ui/fragments/PreviewInfoSource.kt
package com.pr0gramm.app.ui.fragments

import com.pr0gramm.app.feed.FeedItem
import com.pr0gramm.app.ui.PreviewInfo

/**
 */
interface PreviewInfoSource {
    /**
     * Returns the preview info for the given feed item.
     */
    fun previewInfoFor(item: FeedItem): PreviewInfo?
}
